Model/Utilities/WebKitHandler.swift

```diff
@@ -81,52 +81,112 @@ final class KiwixURLSchemeHandler: NSObject, WKURLSchemeHandler {
             stopFor(urlSchemeTask.hash)
             return
         }
-        guard let content = await readContent(for: url) else {
+        guard let metaData = await contentMetaData(for: url) else {
             sendHTTP404Response(urlSchemeTask, url: url)
             return
         }
-        sendHTTP200Response(urlSchemeTask, url: url, content: content)
 
+        let dataProvider: any DataProvider<URLContent>
+        let rangeSize: UInt // The size of the data chunks used for streaming data
+        if metaData.isVideo, let directAccess = await directAccessInfo(for: url) {
```

Comment:
There is no fallback should directAccessInfo return an empty pair. You wisely handled the nul pair in `getDirectAccessInfo` but you're not leveraging it here.
Here we could gracefully fallback to the `ZimContentProvider` should the directAccessInfo be `nil`

Reply:
Hmm, I am not sure if I understand your point fully, I think what you are suggesting is already handled.

In swift: `if let directAccess = <some optional>` will only be `true` when the optional is not `nil`, if it's true, it will assign the value to the variable: `directAccess`  at the same time.
Whereas `if condition == true, otherCondition == true` is the same as: `if condition == true && otherCondition == true`.
If it is `nil` (or it's not a video), it will go into the `else` branch, where we have the `ZimContentProvider` (and in this `else` scope the variable `directAccess` is not available).

After thinking about this a little more, I made some changes.
I moved the validation to the optional (and only) init of the `DirectAccessInfo`. This makes sure we cannot create an invalid instance of it.
On the other hand, I think validating only on the `path` should be enough.
If we have a valid path with offset=0, that should be OK.
On the other-hand if we have an empty path regardless of the offset value, it will be invalid, as we cannot open the file.